src/scripts/destinationList.js

```diff
@@ -173,72 +196,94 @@ let destinationsList = [
 
   //Wishlist
 
-  {
+  paris123: {
+    id: "paris123",
     city: "Paris",
     country: "France",
     photo: require("../img/paris.jpg"),
     visited: false,
   },
-  {
+  santorini123: {
+    id: "santorini123",
     city: "Santorini",
     country: "Greece",
     photo: require("../img/santorini.jpg"),
     visited: false,
   },
-  {
+  newyork123: {
+    id: "newyork123",
     city: "New York",
     country: "U.S.",
     photo: require("../img/newyork.jpg"),
     visited: false,
   },
-  {
+  marrakech: {
+    id: "marrakech",
     city: "Marrakech",
     country: "Morocco",
     photo: require("../img/marrakech.jpg"),
     visited: false,
   },
-  {
+  amsterdam123: {
+    id: "amsterdam123",
     city: "Amsterdam",
     country: "Netherlands",
     photo: require("../img/Amsterdam.png"),
     visited: false,
   },
-  {
+  berlin123: {
+    id: "berlin123",
     city: "Berlin",
     country: "Germany",
     photo: require("../img/berlin.jpg"),
     visited: false,
   },
-  {
+  lisbon123: {
+    id: "lisbon123",
     city: "Lisbon",
     country: "Porugal",
     photo: require("../img/lisbon.jpg"),
     visited: false,
   },
-  {
+  tokyo123: {
+    id: "tokyo123",
     city: "Tokyo",
     country: "Japan",
     photo: require("../img/tokyo.jpg"),
     visited: false,
   },
-  {
+  sienna123: {
+    id: "sienna123",
     city: "Sienna",
     country: "Italy",
     photo: require("../img/sienna.jpg"),
     visited: false,
   },
-  {
+  dublin123: {
+    id: "dublin123",
     city: "Dublin",
     country: "Ireland",
     photo: require("../img/dublin.jpg"),
     visited: false,
   },
-  {
+  petersburg123: {
+    id: "petersburg123",
     city: "Saint petersburg",
     country: "Russia",
     photo: require("../img/saintpetersburg.jpg"),
     visited: false,
   },
-];
+};
+
+const destinationsList = Object.keys(destinationsObj).map((id) => {
+  return destinationsObj[id];
+  // console.log(destinationsObj[id]);
+});
 
 export default destinationsList;
+
+//push new list item to destinationList
+export function addDestination(newItem) {
```

Comment:
> I couldn't find a way to separate them in a way that would follow the logic of the module. Any hints? 😄

So, my recommendation is to go a step further with a new refactor:

1. This file could be renamed to `store.js`. It's a common name when we deal with an app state management. Where we save things is called "stored".

2. In order to update the status of one of the items, those items needs to be stored too. So my recommendation, to start easy, is to ignore `destinationsObj`. Comment all that code, so that it returns an empty obj list. From now on we'll deal only with items manually added in the UI (stored in LS). This will make things easier. Later we can integrate again these static `destinations`. 

3. Remember when I suggested moving the destinations mutation to this file? That's why we created `addDestination` methods. Let's do that with other methods that mutate the list too. For example, move `getStoredList()` from `list.js` to this file, so that all destination items are together (the static and the stored), as well any modifications we do to it. Then the const `destinationsList` will return both static (empty list) and stored lists.

4. Finally, to edit a destination, you need to create a new method `toggleDestinationVisited` that receives the destination id and mutates the localStorage list with the new visited status. Then it also needs to do a pubsub with the new list so the renders updates the list.

Give it a try and push the code with your try! I'll help you through it. Don't forget to use console log to follow the data along, or even go back to paper and pen to draw the data flow diagram :)

Reply:
So ```toggleDestinationVisited``` would be where the elements are created in store.js? 
Out of curiosity, can't the way I started be made to work? To me it looks like it's a step or two away (although messy) 😄

Reply:
`toggleDestinationVisited` would be in `store.js`, exactly!

Regarding your current solution, you don't need to throw it away, it's just incompleted. Some stuff missing:
- the eventListener to checkbox, the classist and status need to be dynamic based on the checkbox state (`event.target.checked`). 
- the LS item `newVisited` does not need to be created (and won't help in anything). Instead, you need to modify the  LS item `destinations`. Instead of adding a new item, to modify an existing one.
I hope this helps, good luck!

Reply:
I pushed 1-3. of the requirements, it doesn't render and I'm not sure why. Here's a diagram, does it make sense? (file names are where they're called)
<img width="1105" alt="Screenshot 2020-10-15 at 10 20 34" src="https://user-images.githubusercontent.com/62510606/96096388-15672600-0ed0-11eb-8d9c-2b932ec3800c.png">

Reply:
Okay, the first mistake was right the first one. I meant to rename `destinationList.js` (where the `destinationObj` is) to `store.js`. You renamed the `list.js` to `store.js` and change `render` methods from place. That wouldn't work...

So, from there, the refactor is broken and that's why stuff don't load. It happens, it's okay. The easiest way is to undo that commit and try again

Make sure the repository has no changes unstaged and the last commit is the _5669b0b_.


```
// revert the commit pushed
git reset HEAD~1
```

You see the change unstaged. Go to VSCode and revert them. [screenshot](https://user-images.githubusercontent.com/14869087/96225461-055f4d00-0f89-11eb-9833-e8358371cc78.png)

Finally push it as it is (with the changes discarded). You need to force the push because you are deleting a commit (which is considered a dangerous git action).

```
git push -f
```

After that, try again. This time here's a flow diagram to help you (I'm doing it, sending in a few minutes)

Reply:
I don't have the option to revert, did I miss something? <img width="1038" alt="Screenshot 2020-10-16 at 17 52 12" src="https://user-images.githubusercontent.com/62510606/96280506-56e2f880-0fd8-11eb-8131-940b28380c4c.png">